pkg/speaker/vip/keepalived.go

```diff
@@ -0,0 +1,192 @@
+package vip
+
+import (
+	"context"
+	"fmt"
+	"github.com/go-logr/logr"
+	"github.com/openelb/openelb/pkg/constant"
+	"github.com/openelb/openelb/pkg/speaker"
+	appv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"strings"
+)
+
+type KeepAlive struct {
+	log       logr.Logger
+	clientset *clientset.Clientset
+	cm        *corev1.ConfigMap
+	conf      *KeepAliveConfig
+}
+
+type KeepAliveConfig struct {
+	Args  []string
+	Image string
+}
+
+func (k *KeepAlive) SetBalancer(configMap string, nexthops []corev1.Node) error {
+	ip := strings.SplitN(configMap, ":", 2)
+	k.cm = &corev1.ConfigMap{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ConfigMap",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      constant.OpenELBConfigMap,
+			Namespace: constant.OpenELBNamespace,
+		},
```

Comment:
The `namespace` should be obtained from the environment variable `OPENELB_NAMESPACE`. The same is true for other places where namespaces are used.